napytau/gui/components/logger.py

```diff
@@ -1,24 +1,121 @@
 import customtkinter
 
+from collections import deque
 from typing import TYPE_CHECKING
 
 if TYPE_CHECKING:
     from napytau.gui.app import App  # Import only for the type checking.
 
+# Define color constants for the logger
+LIGHT_MODE_INFO_COLOR = "black"
+LIGHT_MODE_ERROR_COLOR = "#ba1500"
+LIGHT_MODE_SUCCESS_COLOR = "#388f00"
+
+DARK_MODE_INFO_COLOR = "white"
+DARK_MODE_ERROR_COLOR = "#ff3a21"
+DARK_MODE_SUCCESS_COLOR = "#48ba00"
+
 
 class Logger(customtkinter.CTkFrame):
     def __init__(self, parent: "App") -> None:
         """
-        Initializes the logger frame.
+        The logger for the GUI.
         :param parent: Parent widget to host the logger.
         """
-        super().__init__(parent, height=100, corner_radius=10)
+        super().__init__(parent, height=10, corner_radius=10)
         self.parent = parent
+
+        self.grid(row=2, column=0, columnspan=1, padx=(10, 10), pady=(10, 10),
+                  sticky="ew")
         self.grid_propagate(False)
-        self.grid(row=2, column=0, columnspan=2, padx=10, pady=10, sticky="nsew")
 
-        # Label to display messages
-        self.output_label = customtkinter.CTkLabel(
-            self, text="Error messages etc. will be shown here", anchor="w"
+        # Create a scrollable frame for the log messages
+        self.scrollable_frame = customtkinter.CTkScrollableFrame(self, height=40)
+        self.scrollable_frame.pack(fill="both", expand=True, padx=10, pady=10)
+
+        # Set initial text colors
+        if customtkinter.get_appearance_mode() == "Light":
+            self.info_color = LIGHT_MODE_INFO_COLOR
+            self.error_color = LIGHT_MODE_ERROR_COLOR
+            self.success_color = LIGHT_MODE_SUCCESS_COLOR
+        else:
+            self.info_color = DARK_MODE_INFO_COLOR
+            self.error_color = DARK_MODE_ERROR_COLOR
+            self.success_color = DARK_MODE_SUCCESS_COLOR
+
+        # Store labels in a deque with a max length of 50
+        self.labels: deque = deque(maxlen=50)
```

Comment:
Why is a deque necessary here, doesn't a queue suffice?

Reply:
The deque should be more efficient because with a normal queue you cannot really edit items while iterating.
You would have to do something like this:
```
temporary_labels = []
while not self.labels.empty():
	label = self.labels.get()
	temporary_labels.append(label)
	# Configure label
	
for label in temporary_labels:
	self.labels.put(label)
```
instead of this:
```
for label in self.labels:
# Configure label
```

Reply:
Hmm, im not that deep into python data structures and this sounds reasonable. Keep it as is then, but add a small comment summarizing what you explained here.